### PR TITLE
Fix transient mounts in Docker 1.12

### LINKED ISSUE #6: starting container with HostConfig was deprecated since v1.10 and removed in v1.12
On a machine with docker v1.12 it throws above error error, but works fine on docker v1.10.3

running:
```bash
$ imagebuilder . surajkanginx
--> Image fedora was not found, pulling ...
--> Pulled 0/1 layers, 7% complete
--> Pulled 1/1 layers, 100% complete
--> Extracting
--> FROM fedora
unable to start build container: API error (400): {"message":"starting container with HostConfig was deprecated since v1.10 and removed in v1.12"}
```

dockerfile:
```bash
 $ cat Dockerfile 
FROM fedora
MAINTAINER http://fedoraproject.org/wiki/Cloud

RUN dnf -y update && dnf clean all
RUN dnf -y install nginx && dnf clean all
RUN echo "daemon off;" >> /etc/nginx/nginx.conf
RUN echo "nginx on Fedora" > /usr/share/nginx/html/index.html

EXPOSE 80

CMD [ "/usr/sbin/nginx" ]
```

docker version
```bash
$ docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        
 OS/Arch:      linux/amd64
```

### PR DESCRIPTION
**Pull request by smarterclayton**
Use a temporary container instead of hostconfig on start.  Add a test case.

Fixes #6

### CODE REVIEW

**Comment by smarterclayton** [comment]
@jawnsy can you give this a try on 1.12 and see it works for you?

**Comment by smarterclayton** [comment]
Good news - there appears to be a race in docker 1.12 where volumes aren't initialized prior to startup.  That's what blocks this.

File: README.md [hunk 1]
```diff
@@ -40,10 +40,11 @@ To download and install the library and the binary, set up a Golang build enviro
 $ go get -u github.com/openshift/imagebuilder/cmd/imagebuilder
 ```
 
-The included command line takes two arguments, a path to a directory containing a dockerfile, and the name to tag the output image with.
+The included command line takes one argument, a path to a directory containing a Dockerfile. The `-t` option
+can be used to specify an image to tag as.
 
 ```
-$ imagebuilder path_to_directory output_image_name
+$ imagebuilder [-t TAG] DIRECTORY
```
Comment (jawnsy) [inline README.md]: Isn't this a breaking change? What's our compatibility statement with respect to use of `imagebuilder` in scripts?

Comment (smarterclayton) [inline README.md]: Yes, and none (yet).

**Comment by smarterclayton** [comment]
Merging to unbreak docker 1.12 for everything except transient mounts.

**Pull request merged**